### PR TITLE
Don't require `setuptools_rust` at runtime

### PR DESCRIPTION
**Pull request by DMRobertson**
Fixes https://github.com/matrix-org/synapse/issues/13926, see https://github.com/matrix-org/synapse/issues/13926#issuecomment-1260740889

I don't like special-casing it, but :shrug:

### CODE REVIEW

**Comment by DMRobertson** [comment]
@erikjohnston, does this suggest we have been including `setuptools_rust` in our docker images and debian packages at runtime too?

File: synapse/util/check_dependencies.py [hunk 1]
```diff
@@ -66,6 +66,17 @@ def _is_dev_dependency(req: Requirement) -> bool:
     )
 
 
+def _should_ignore_runtime_requirement(req: Requirement) -> bool:
+    # This is a build-time dependency. Irritatingly, `poetry build` ignores the
+    # requirements listed in the [build-system] section of pyproject.toml, so in order
+    # to support `poetry install --no-dev` we have to mark it as a runtime dependency.
+    # Workaround this by ignoring it here. (It might be slightly cleaner to put
+    # `setuptools_rust` in a `build` extra or similar, but . But for now I'
```
Comment (squahtx) [inline synapse/util/check_dependencies.py:74]: some words have gotten lost here

**Review by squahtx — APPROVED** [review]

File: synapse/util/check_dependencies.py [hunk 2]
```diff
@@ -66,6 +66,17 @@ def _is_dev_dependency(req: Requirement) -> bool:
     )
 
 
+def _should_ignore_runtime_requirement(req: Requirement) -> bool:
+    # This is a build-time dependency. Irritatingly, `poetry build` ignores the
+    # requirements listed in the [build-system] section of pyproject.toml, so in order
```
Comment (clokep) [inline synapse/util/check_dependencies.py:71]: Should we link to a poetry bug here?

Comment (clokep) [reply synapse/util/check_dependencies.py:71]: https://github.com/python-poetry/poetry/issues/6154 I guess would be this and is linked to someplace else, so probably OK not to.

File: synapse/util/check_dependencies.py (hunk 1 above)
Comment (DMRobertson) [reply synapse/util/check_dependencies.py:74]: :facepalm: Guh, I think this is a sign I need to eat. Thanks, will amend

File: synapse/util/check_dependencies.py (hunk 2 above)
Comment (DMRobertson) [reply synapse/util/check_dependencies.py:71]: I wasn't 100% sure if this is considered a bug by poetry's authors, but the issue is a good point of reference---I'll gladly add it

**Review by clokep — APPROVED** [review]

**Pull request merged**